### PR TITLE
use domain setting over feature preview

### PR DESCRIPTION
**Pull request by snopoke**
Just something I noticed.
@NoahCarnahan

### CODE REVIEW

File: corehq/apps/domain/forms.py [hunk 1]
```diff
@@ -680,12 +680,9 @@ class DomainMetadataForm(DomainGlobalSettingsForm):
     )
 
     def __init__(self, *args, **kwargs):
-        user = kwargs.pop('user', None)
-        domain = kwargs.get('domain', None)
         super(DomainMetadataForm, self).__init__(*args, **kwargs)
 
-        project = Domain.get_by_name(domain)
-        if project.cloudcare_releases == 'default' or not domain_has_privilege(domain, privileges.CLOUDCARE):
+        if self.project.cloudcare_releases == 'default' or not domain_has_privilege(self.domain, privileges.CLOUDCARE):
```
Comment (dimagimon) [inline corehq/apps/domain/forms.py]: E501 line too long (119 > 115 characters)

**Comment by NoahCarnahan** [comment]
Test failure looks real.
Looks like maybe some `SimpleTestCase`s that call `get_call_center_case_type_if_enabled()` need to mock the `Domain.get_by_name()`.

File: corehq/apps/callcenter/utils.py [hunk 2]
```diff
@@ -233,19 +233,25 @@ def get_call_center_domains():
...
-            name=hit['name'],
-            default_timezone=hit['default_timezone'],
-            cc_case_type=config.get('case_type', ''),
-            use_fixtures=config.get('use_fixtures', True)
-        )
-    return [to_domain_lite(hit) for hit in result.hits]
+        case_type = config.get('case_type', None)
+        case_owner_id = config.get('case_owner_id', None)
+        use_user_location_as_owner = config.get('use_user_location_as_owner', None)
+        if case_type and (case_owner_id or use_user_location_as_owner):  # see CallCenterProperties.config_is_valid()
```
Comment (dimagimon) [inline corehq/apps/callcenter/utils.py]: E501 line too long (117 > 115 characters)

File: corehq/apps/callcenter/tests/test_utils.py [hunk 3]
```diff
@@ -339,14 +339,17 @@ def setUp(self):
         self.elasticsearch = get_es_new()
         ensure_index_deleted(self.index_info.index)
         initialize_index(self.elasticsearch, self.index_info)
+        import time
+        time.sleep(1)  # without this we get a 503 response about 30% of the time
```
Comment (snopoke) [inline corehq/apps/callcenter/tests/test_utils.py]: not super happy about this

**Comment by snopoke** [comment]
updated and added a few commits

**Comment by millerdev** [comment]
👍 on ✅

**Comment by dannyroberts** [comment]
```
======================================================================
ERROR: test_basic_application_schema (corehq.apps.export.tests.test_export_data_schema.TestBuildingCaseSchemaFromApplication)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/apps/export/tests/test_export_data_schema.py", line 463, in test_basic_application_schema
    schema = CaseExportDataSchema.generate_schema_from_builds(self.domain, self.case_type)
  File "/mnt/corehq/apps/export/models/new.py", line 1049, in generate_schema_from_builds
    include_parent_properties=False
  File "/mnt/corehq/apps/app_manager/util.py", line 287, in get_case_properties
    per_type_defaults = get_per_type_defaults(app.domain, case_types)
  File "/mnt/corehq/apps/app_manager/util.py", line 305, in get_per_type_defaults
    callcenter_case_type = get_call_center_case_type_if_enabled(domain)
  File "/mnt/corehq/util/quickcache.py", line 312, in inner
    return helper(*args, **kwargs)
  File "/mnt/corehq/util/quickcache.py", line 100, in __call__
    content = self.fn(*args, **kwargs)
  File "/mnt/corehq/apps/callcenter/utils.py", line 281, in get_call_center_case_type_if_enabled
    config = domain_object.call_center_config
AttributeError: 'NoneType' object has no attribute 'call_center_config'
```
and

```
======================================================================
ERROR: test_simple_case_management (corehq.apps.userreports.tests.test_app_manager_integration.AppManagerDataSourceConfigTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/mnt/corehq/apps/userreports/tests/test_app_manager_integration.py", line 33, in test_simple_case_management
    data_sources = get_case_data_sources(app)
  File "/mnt/corehq/apps/userreports/app_manager.py", line 22, in get_case_data_sources
    return {case_type: get_case_data_source(app, case_type) for case_type in app.get_case_types() if case_type}
  File "/mnt/corehq/apps/userreports/app_manager.py", line 22, in <dictcomp>
    return {case_type: get_case_data_source(app, case_type) for case_type in app.get_case_types() if case_type}
  File "/mnt/corehq/apps/userreports/app_manager.py", line 27, in get_case_data_source
    prop_map = get_case_properties(app, [case_type], defaults=DEFAULT_CASE_PROPERTY_DATATYPES.keys())
  File "/mnt/corehq/apps/app_manager/util.py", line 287, in get_case_properties
    per_type_defaults = get_per_type_defaults(app.domain, case_types)
  File "/mnt/corehq/apps/app_manager/util.py", line 305, in get_per_type_defaults
    callcenter_case_type = get_call_center_case_type_if_enabled(domain)
  File "/mnt/corehq/util/quickcache.py", line 312, in inner
    return helper(*args, **kwargs)
  File "/mnt/corehq/util/quickcache.py", line 100, in __call__
    content = self.fn(*args, **kwargs)
  File "/mnt/corehq/apps/callcenter/utils.py", line 280, in get_call_center_case_type_if_enabled
    domain_object = Domain.get_by_name(domain)
  File "/mnt/corehq/util/quickcache.py", line 312, in inner
    return helper(*args, **kwargs)
  File "/mnt/corehq/util/quickcache.py", line 201, in __call__
    return super(SkippableQuickCache, self).__call__(*args, **kwargs)
  File "/mnt/corehq/util/quickcache.py", line 100, in __call__
    content = self.fn(*args, **kwargs)
  File "/mnt/corehq/apps/domain/models.py", line 538, in get_by_name
    domain = _get_by_name(stale=(not strict))
  File "/mnt/corehq/apps/domain/models.py", line 530, in _get_by_name
    result = cls.view("domain/domains", key=name, reduce=False, include_docs=True, **extra_args).first()
  File "/vendor/lib/python2.7/site-packages/couchdbkit/schema/base.py", line 310, in view
    return db.view(view_name,
  File "/vendor/lib/python2.7/site-packages/mock.py", line 700, in __getattr__
    raise AttributeError("Mock object has no attribute %r" % name)
AttributeError: Mock object has no attribute 'view'
```

**Pull request merged**